e2e/terraform/shared/scripts/setup.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+set -e
+
+# Disable interactive apt prompts
+export DEBIAN_FRONTEND=noninteractive
+
+cd /ops
+
+CONFIGDIR=/ops/shared/config
+
+CONSULVERSION=1.3.1
+CONSULDOWNLOAD=https://releases.hashicorp.com/consul/${CONSULVERSION}/consul_${CONSULVERSION}_linux_amd64.zip
+CONSULCONFIGDIR=/etc/consul.d
+CONSULDIR=/opt/consul
+
+VAULTVERSION=0.11.4
+VAULTDOWNLOAD=https://releases.hashicorp.com/vault/${VAULTVERSION}/vault_${VAULTVERSION}_linux_amd64.zip
+VAULTCONFIGDIR=/etc/vault.d
+VAULTDIR=/opt/vault
+
+NOMADVERSION=0.8.6
+NOMADDOWNLOAD=https://releases.hashicorp.com/nomad/${NOMADVERSION}/nomad_${NOMADVERSION}_linux_amd64.zip
+NOMADCONFIGDIR=/etc/nomad.d
+NOMADDIR=/opt/nomad
+
+HADOOP_VERSION=2.7.6
+
+# Dependencies
+sudo apt-get install -y software-properties-common
+sudo apt-get update
+sudo apt-get install -y unzip tree redis-tools jq curl tmux awscli
+
+# Numpy (for Spark)
+sudo apt-get install -y python-setuptools
+sudo easy_install pip
+sudo pip install numpy
+
+# Disable the firewall
+
+sudo ufw disable || echo "ufw not installed"
+
+# Consul
+
+curl -L $CONSULDOWNLOAD > consul.zip
+
+## Install
+sudo unzip consul.zip -d /usr/local/bin
+sudo chmod 0755 /usr/local/bin/consul
+sudo chown root:root /usr/local/bin/consul
+
+## Configure
+sudo mkdir -p $CONSULCONFIGDIR
+sudo chmod 755 $CONSULCONFIGDIR
+sudo mkdir -p $CONSULDIR
+sudo chmod 755 $CONSULDIR
+
+# Vault
+
+curl -L $VAULTDOWNLOAD > vault.zip
+
+## Install
+sudo unzip vault.zip -d /usr/local/bin
+sudo chmod 0755 /usr/local/bin/vault
+sudo chown root:root /usr/local/bin/vault
+
+## Configure
+sudo mkdir -p $VAULTCONFIGDIR
+sudo chmod 755 $VAULTCONFIGDIR
+sudo mkdir -p $VAULTDIR
+sudo chmod 755 $VAULTDIR
+
+# Nomad
+
+curl -L $NOMADDOWNLOAD > nomad.zip
```

Comment:
We should probably remove the Nomad install in this step even though we end up over-writing it in Terraform at run-time.